scripts/get_objf_and_derivs.py

```diff
@@ -249,6 +255,26 @@ def WriteDerivs():
         print("order{0}_D4 {1}".format(o, d4_deriv[o]), file=f)
     f.close()
 
+def CleanUp():
+    if args.clean_up == 'true':
+        try:
+            if not os.path.isdir(args.work_dir):
+                ExitProgram("error finding working directory {0}".format(args.work_dir))
+            for file in os.listdir(args.work_dir):
+                need_float='\d+' if args.need_model == 'true' else ''
+                if re.match('(discounted|discounted_derivs|float|float_derivs|merged|merged_derivs)\.'+need_float,file):
```

Comment:
Thanks, Dan
I fixed that
but cluster seems to have some problem to login? When I can login, I will do the test then re-do pull request

On Jul 11, 2016, at 3:26 PM, Daniel Povey <notifications@github.com> wrote:

> In scripts/get_objf_and_derivs.py:
> 
> > @@ -249,6 +255,26 @@ def WriteDerivs():
> >          print("order{0}_D4 {1}".format(o, d4_deriv[o]), file=f)
> >      f.close()
> >  
> > +def CleanUp():
> > +    if args.clean_up == 'true':
> > +        try:
> > +            if not os.path.isdir(args.work_dir):
> > +                ExitProgram("error finding working directory {0}".format(args.work_dir))
> > +            for file in os.listdir(args.work_dir):
> > +                need_float='\d+' if args.need_model == 'true' else ''
> > +                if re.match('(discounted|discounted_derivs|float|float_derivs|merged|merged_derivs)\.'+need_float,file):
> This doesn't seem a very exact way of excluding the removal of the 'float.all' file, and it's also not very clear.
> Please just compre float.all by name before you remove it, this will be clearer.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>